pkg/kubectl/cmd/set/set_env.go

```diff
@@ -173,18 +181,66 @@ func keyToEnvName(key string) string {
 	return strings.ToUpper(validEnvNameRegexp.ReplaceAllString(key, "_"))
 }
 
-func (o *EnvOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
-	if o.All && len(o.Selector) > 0 {
-		return fmt.Errorf("cannot set --all and --selector at the same time")
+func contains(key string, keyList []string) bool {
+	if len(keyList) == 0 {
+		return true
 	}
-	resources, envArgs, ok := envutil.SplitEnvironmentFromResources(args)
-	if !ok {
-		return cmdutil.UsageErrorf(o.Cmd, "all resources must be specified before environment changes: %s", strings.Join(args, " "))
+
+	for _, k := range keyList {
+		if k == key {
+			return true
+		}
 	}
-	if len(o.Filenames) == 0 && len(resources) < 1 {
-		return cmdutil.UsageErrorf(cmd, "one or more resources must be specified as <resource> <name> or <resource>/<name>")
+	return false
+}
+
+func (o *EnvOptions) getConfigResources(f cmdutil.Factory, cmd *cobra.Command) ([]*v1.ConfigMap, []*v1.Secret, error) {
```

Comment:
Do we plan on this being an ever-expanding list of resources that we support getting env values from?
If not, just rename this to something like `getConfimapsAndSecrets` or something similar.

If we do plan on growing the list of supported resources, I think I'd rather we return `[]runtime.Object`. It would still only contain the resources of kinds that are supported, but the caller of this method could sort those out. @soltysh for an additional opinion on this.

Reply:
Given your comment below about aggregating on the struct I guess I can just change this function to a setter and return an error only, wdyt?

Reply:
Sure, please move the call to this method to Complete also. Thanks